src/util/ThreadSafeQueue.h

```diff
@@ -16,11 +16,39 @@
 
 #include "util/Exception.h"
 #include "util/ExceptionHandling.h"
-#include "util/Generator.h"
+#include "util/Iterators.h"
 #include "util/jthread.h"
 
-namespace ad_utility::data_structures {
+namespace {
+
+template <typename Queue, typename Producer, typename MakeQueueTask>
+struct QueueGenerator
```

Comment:
I think you can define this as a local class directly inside the `queueManager` where you then don't need (and also can't use) the template arguments, because you interit them from the function.